lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -144,10 +144,26 @@ function isObjectLocked(bucket, objectMD, headers) {
     return false;
 }
 
+function validateObjectLockUpdate(objectMD, retentionInfo) {
+    const { retentionMode: existingMode, retentionDate: existingDate } = objectMD;
```

Comment:
could existing date string be `existingDateISO` so it is clear it is not a Date already?